### ISSUE TITLE
Model does not converge when running model_tutorial.ipynb (loss increases after training)

### ISSUE THREAD

**Issue by zz310** [snap-stanford/GEARS#96]
Hi, thanks for releasing this great project!

I was trying to reproduce the results by running `model_tutorial.ipynb` following the default settings.
However, I noticed that after 20 epochs of training, the model does not converge — in fact, the training loss slightly increases.

- **Initial training loss:** ~0.47  
- **Final training loss (after 20 epochs):** ~0.51  

Also, When I load the Norman dataset, I see multiple warnings like:
These perturbation are not in the GO graph and their perturbation can thus not be predicted.

Can I know if these warning are expected?
Thanks for your help!

**Comment by linjing-lab** [comment]
The loss function in machine learning training increases because it is a module defined by MLP generative model. Therefore, when the model parameters are not complex, too many iterations lead to the model jumping out of the region of local convergence, resulting in a slight increase in model loss. When I studied my multimodal spatial omics integration, I also found that the loss of spectral clustering increased, but the overall loss decreased.
